src/sentry/api/authentication.py

```diff
@@ -127,9 +132,59 @@ def relay_from_id(request, relay_id) -> Tuple[Optional[Relay], bool]:
 
 
 class QuietBasicAuthentication(BasicAuthentication):
+    _hybrid_cloud_rollout_level = -1
+
     def authenticate_header(self, request: Request) -> str:
         return 'xBasic realm="%s"' % self.www_authenticate_realm
 
+    def _use_authenticated_token(self) -> bool:
+        return (
+            SiloMode.get_current_mode() != SiloMode.MONOLITH
+            or options.get("hybrid_cloud.authentication.use_authenticated_token")
+            >= self._hybrid_cloud_rollout_level
+            or in_test_environment()
+        )
+
+    def _use_rpc_user(self) -> bool:
+        return (
+            SiloMode.get_current_mode() != SiloMode.MONOLITH
+            or options.get("hybrid_cloud.authentication.use_rpc_user")
+            >= self._hybrid_cloud_rollout_level
+            or in_test_environment()
+        )
+
+    def transform_auth(
+        self,
+        user: int | User | RpcUser | None | AnonymousUser,
+        auth: Any,
+        entity_id_tag: str | None = None,
+        **tags: Any,
+    ) -> Tuple[User | RpcUser | AnonymousUser, Any]:
+        if isinstance(user, int):
+            if self._use_rpc_user():
+                user = user_service.get_user(user_id=user)
+            else:
+                user = User.objects.filter(id=user).first()
+        elif isinstance(user, User):
+            if self._use_rpc_user():
+                user = user_service.get_user(user_id=user.id)
+        if user is None:
+            user = AnonymousUser()
+
+        auth_token = AuthenticatedToken.from_token(auth)
+        if auth_token and entity_id_tag:
+            with configure_scope() as scope:
+                scope.set_tag(entity_id_tag, auth_token.entity_id)
+                for k, v in tags.items():
+                    scope.set_tag(k, v)
+
+        return (
+            user,
+            AuthenticatedToken.from_token(auth)
```

Comment:
Are we able to reuse `auth_token` given that we already made one?

Reply:
Fixed